torch/csrc/jit/mobile/type_parser.cpp

```diff
@@ -160,11 +198,37 @@ class TypeParser {
   std::string pythonStr_;
   size_t start_;
   std::string next_token_;
+  // Store all contained types when parsing a string
+  std::unordered_set<std::string> contained_types_;
 };
 } // namespace
 
 TORCH_API TypePtr parseType(const std::string& pythonStr) {
-  TypeParser paser(pythonStr);
-  return paser.parse();
+  TypeParser parser(pythonStr);
+  return parser.parse();
 }
+
+// Get all contained type given a string
+TORCH_API std::unordered_set<std::string> getContainedTypes(
```

Comment:
Would those be used in compatibility API only? Could you put them in a different file, since it does not belong to type parsing itself?

Reply:
TypeParser is declared in `type_parser.cpp` only, and other cpp can't access it. Move the class definition to header file is an option. Do you see any issue?